### PR TITLE
Add playbooks for cloud provider security groups

### PR DESCRIPTION
**Pull request by edwardzsmith**

### CODE REVIEW

File: aws/add-security-group-rule/add-security-group-rule.yml [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+            - Provide a rule description
+
+      - ppa.ui.input_accept:
+          text: Start
+
+      - ppa.ui.input_choice:
+          text: Select an EC2 Region
+        load:
+          options: regions
+        save: region
```
Comment (NeilGerring) [inline aws/add-security-group-rule/add-security-group-rule.yml:33]: Not a big issue at all. But I wonder if this should be done after attempting to get the secrets?

File: aws/add-security-group-rule/add-security-group-rule.yml [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+
+      - eval:
+          expression: len(groups)
+        save: group_count
+
+      - ppa.ui.output_progress:
+          text: Found {{ group_count }} Security Groups
+          percent: 100
+          element_id: get_groups
+
```
Comment (NeilGerring) [inline aws/add-security-group-rule/add-security-group-rule.yml:71]: I'd be tempted to move this into it's own step

File: aws/add-security-group-rule/add-security-group-rule.yml [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+          kv_version: 1
+        save: aws_secrets
+
+      - ppa.ui.output_progress:
+          text: Finding all Security Groups...
+          percent: null
+          element_id: get_groups
+
+      - aws.ec2.security_groups.get_all:
+          region_name: eu-west-2
```
Comment (NeilGerring) [inline aws/add-security-group-rule/add-security-group-rule.yml:53]: Should this be hard coded? I think it should be using the previously selected region

File: aws/add-security-group-rule/add-security-group-rule.yml [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+      - aws.ec2.security_groups.input_table:
+          text: Select a Security Group
+          minimum: 1
+          maximum: 1
+        load:
+          groups: groups
+        save: chosen_group
+
+      - debug:
+          identifier: chosen_group
```
Comment (NeilGerring) [inline aws/add-security-group-rule/add-security-group-rule.yml:83]: This should probably be removed

File: aws/add-security-group-rule/add-security-group-rule.yml [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+    actions:
+      - ppa.ui.input_choice:
+          text: Select a Port
+          options:
+            HTTP (80):
+              order: 1
+              value: 80
+            HTTPS (443):
+              order: 1
+              value: 443
```
Comment (NeilGerring) [inline aws/add-security-group-rule/add-security-group-rule.yml:114]: Should these both have an order of `1`?

File: aws/add-security-group-rule/add-security-group-rule.yml [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+
+  - name: Create Rule
+    actions:
+      - ppa.ui.output_progress:
+          text: Creating new Security Group inbound rule...
+          percent: null
+          element_id: create_rule
+
+      - aws.ec2.security_groups.create_inbound_rule:
+          region_name: eu-west-2
```
Comment (NeilGerring) [inline aws/add-security-group-rule/add-security-group-rule.yml:174]: Should this be hardcoded?

File: aws/add-security-group-rule/add-security-group-rule.yml [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+        save: view_updated_rules
+
+      - exit:
+          exit_code: 0
+        when: not view_updated_rules
+
+  - name: View Updated Rules
+    actions:
+      - aws.ec2.security_groups.get_by_name:
+          region_name: eu-west-2
```
Comment (NeilGerring) [inline aws/add-security-group-rule/add-security-group-rule.yml:202]: Should this be hardcoded?

File: aws/add-security-group-rule/add-security-group-rule.yml (hunk 1 above)
Comment (edwardzsmith) [reply aws/add-security-group-rule/add-security-group-rule.yml:33]: yes!

**Pull request merged**